system/src/system_control_internal.cpp

```diff
@@ -389,6 +407,28 @@ void system_ctrl_set_result(ctrl_request* req, int result, ctrl_completion_handl
     particle::system::SystemControl::instance()->setResult(req, result, handler, data);
 }
 
+int system_ctrl_set_request_filter(uint16_t reqType, void* reserved) {
```

Comment:
`system_ctrl_add_request_filter` is more accurate. "set" sounds like we're going to remove existing elements. Alternatively, pass in an array and element count as its argument to set the filter for once.